torch/testing/_internal/common_pruning.py

```diff
@@ -7,6 +7,10 @@
 from torch import nn
 
 
+def elements_are_subset(subset_tensor, superset_tensor):
```

Comment:
I didn't follow the rest of your code, but I assume pruning here means that certain "rows" are simply removed from the superset_tensor. If that's the case, you can search for the `subset_tensor` as you sequentially sweep through `superset_tensor` to avoid a lot of duplicate comparisons (and also avoid some edge case false True returns, i.e. matching the same row multiple times):

```
def elements_are_subset_ordered(subset_tensor, superset_tensor):
    i = 0
    for row in subset_tensor:  # or iterate over whichever dimension is appropriate
        while i < len(superset_tensor):
            if not torch.equal(row, superset_tensor[i]):
                i += 1  # keep searching
            else:
                break  # check next row (continue for loop)
        else:
            return False  # searched beyond end of superset_tensor

    return True  # matched all in subset_tensor
```

I'm happy to discuss if I misunderstood the assumptions here.